velox/benchmarks/tpch/TpchBenchmark.cpp

```diff
@@ -155,21 +156,74 @@ int main(int argc, char** argv) {
     folly::runBenchmarks();
   } else {
     const auto queryPlan = queryBuilder->getQueryPlan(FLAGS_run_query_verbose);
-    const auto task = benchmark.run(queryPlan);
-    ensureTaskCompletion(task.get());
-    const auto stats = task->taskStats();
-    std::cout << fmt::format(
-                     "Execution time: {}",
-                     succinctMillis(
-                         stats.executionEndTimeMs - stats.executionStartTimeMs))
-              << std::endl;
-    std::cout << fmt::format(
-                     "Splits total: {}, finished: {}",
-                     stats.numTotalSplits,
-                     stats.numFinishedSplits)
-              << std::endl;
-    std::cout << printPlanWithStats(
-                     *queryPlan.plan, stats, FLAGS_include_custom_stats)
-              << std::endl;
+    std::vector<double> executionTime;
+
+    for (int iter = 0; iter <= FLAGS_num_runs; iter++) {
+      const auto task = benchmark.run(queryPlan);
+      ensureTaskCompletion(task.get());
+      const auto stats = task->taskStats();
+
+      if (iter == 0) {
+        executionTime.push_back(
+            stats.executionEndTimeMs - stats.executionStartTimeMs);
+
+        std::cout << fmt::format("Iteration: {}\t", iter + 1);
+        std::cout << fmt::format(
+                         "Execution time: {}",
+                         succinctMillis(
+                             stats.executionEndTimeMs -
+                             stats.executionStartTimeMs))
+                  << std::endl;
+        std::cout << fmt::format(
+                         "Splits total: {}, finished: {}",
+                         stats.numTotalSplits,
+                         stats.numFinishedSplits)
+                  << std::endl;
+        std::cout << printPlanWithStats(
+                         *queryPlan.plan, stats, FLAGS_include_custom_stats)
+                  << std::endl;
+      } else {
+        executionTime.push_back(
+            stats.executionEndTimeMs - stats.executionStartTimeMs);
+
+        std::cout << fmt::format("Iteration: {}\t", iter + 1);
+        std::cout << fmt::format(
+            "Execution time: {}\n", succinctMillis(executionTime.back()));
+      }
+    }
+
+    auto minTime = std::min_element(executionTime.begin(), executionTime.end());
+    auto maxTime = std::max_element(executionTime.begin(), executionTime.end());
+    auto avgTime = std::reduce(executionTime.begin(), executionTime.end()) /
+        executionTime.size();
+    auto numObservations = executionTime.size();
+    fmt::print("\nMinimum execution time: {}\n", succinctMillis(*minTime));
+    fmt::print("Maximum execution time: {}\n", succinctMillis(*maxTime));
+    fmt::print("Average execution time: {}\n", succinctMillis(avgTime));
+
+    if (numObservations % 2) {
```

Comment:
This logic can be simplified to completely sort the times and pick the middle number. There will likely be few runs so the partial sorting seems a bit complicated.

Reply:
Thanks for the suggestion, I will change it accordingly.